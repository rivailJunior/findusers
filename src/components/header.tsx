import React from 'react';
import styles from './header.module.scss'

type iHeader = {
    userName?: string,
    labelRight?: string,
    handleRight?(): Function
}
const Header = ({ userName, labelRight, handleRight }: iHeader): JSX.Element => {
    return (
        <div className={styles.header}>
            <a href="#default" className={styles.logo}>{userName}</a>
            {labelRight && (
                <div className={styles['header-right']}>
                    <a className={styles.active} onClick={handleRight} >Sair</a>
                </div>
            )}

        </div>

    );
}

export { Header };
