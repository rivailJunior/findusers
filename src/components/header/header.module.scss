@use '../../pages/index.scss' as index;
@use '../../pages/root.scss' as root;
.header {
    overflow: hidden;
    background-color: white;
    padding: 5px 10px;
    display: grid;
    display: grid;
    grid-template-columns: repeat(12, 1fr);

    .headerLeft {
        grid-column: span 4;
        display: flex;
        align-items: center;
    }

    .headerRight {
        grid-column: span 8;

        div {
            justify-content: flex-end;
            display: flex;
            flex-direction: row;
            width: 100%;
            right: 0;
            align-items: center;
        }
    }

    a {
        display: flex;
        justify-content: center;
        align-items: center;
        float: left;
        color: black;
        text-align: left;
        text-decoration: none;
        font-size: 15px;
        line-height: 25px;
        border-radius: 4px;
    }

    a.logo {
        font-size: 25px;
        font-weight: bold;
        @include index.default-label(18px, 1x, 10, 500, 28px, left);
    }

    a.active {
        background-color: root.$color;
        color: white;
        height: 40px;
        width: 40px;
        @include index.default-shadow();
    }
    .active:hover {
        cursor: pointer;
    }
}

.header-right {
    float: right;
}

/* Smartphones (portrait and landscape) ----------- */
@media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
    .header {
        .headerLeft {
            grid-column: span 12;
        }

        .headerRight {
            grid-column: span 12;

            div {
                justify-content: flex-end;
                display: flex;
                flex-direction: row;
                width: 100%;
                right: 0;
            }
        }
    }
}
