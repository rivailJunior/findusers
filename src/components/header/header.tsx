import React from 'react';
import styles from './header.module.scss'
import { SearchInput } from '../searchInput/searchInput';

type iHeader = {
    userName?: string,
    labelRight?: string,
    handleRight?(): Function
    handleSearch?: Function
}

const Header = ({ userName, labelRight, handleRight, handleSearch }: iHeader): JSX.Element => {
    return (
        <div className={styles.header}>
            <div className={styles.headerLeft}>
                <a href="/" className={styles.logo}>{userName}</a>
            </div>
            <div className={styles.headerRight}>
                <div>
                    {handleSearch && <SearchInput handleSearch={handleSearch} placeholder="Buscar Usuario" />}

                    {labelRight && (
                        <div className={styles['header-right']}>
                            <a className={styles.active} onClick={handleRight} >Sair</a>
                        </div>
                    )}
                </div>

            </div>
        </div>

    );
}

export { Header };
export default Header;
