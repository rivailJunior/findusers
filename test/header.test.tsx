import React from 'react'
import { render, cleanup, screen, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import { Header } from '../src/components/header/header'

afterEach(cleanup)

const fakeFn = jest.fn()
describe('Header Component', () => {
    test('Render Correctly', () => {
        const container = renderer.create(<Header userName="Rivail Junior" labelRight="Sair" handleRight={fakeFn} />)
        container.toJSON();
        expect(container).toMatchSnapshot()
    })

    test('Click on logout button', () => {
        render(<Header userName="Rivail Junior" labelRight="Sair" handleRight={fakeFn} />)
        fireEvent.click(screen.getByText(/Sair/i));
        expect(fakeFn).toHaveBeenCalled()
        expect(fakeFn).toHaveBeenCalledTimes(1)
    })


    test('Hide Search field', () => {
        render(<Header userName="Rivail Junior" labelRight="Sair" />)
        expect(screen.queryByText('Buscar Usuario')).not.toBeInTheDocument()
    })


})