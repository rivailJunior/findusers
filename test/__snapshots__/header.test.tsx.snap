// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Render Correctly 1`] = `
<div
  className="header"
>
  <a
    className="logo"
    href="#default"
  >
    Rivail Junior
  </a>
  <div
    className="header-right"
  >
    <a
      className="active"
      onClick={[MockFunction]}
    >
      Sair
    </a>
  </div>
</div>
`;
